### ISSUE TITLE
Prop is the correct type, but console still indicates failed prop type (Invalid prop `data` of type `array` supplied to `Results`, expected `object`)

### ISSUE THREAD

**Issue by cagross** [yannickcr/eslint-plugin-react#2573]
I've added a typecheck to one of my props, using PropType.  Specifically, I've specified that the prop `data` should be type _object_ when passed to the `Results()` functional component, like so:

```
Results.propTypes = {
	data: PropTypes.object,
  };
```

Later in my script I have added these JSX lines, to print info to the console log, and call the aforementioned `Results()` function:

```
{console.log(typeof(data))}
{console.log(data)}
<Results currMeat={currentMeat} data={data} />
```

On the front-end, the JS console does indeed print out the type of the prop in-question (`data`).  That type is _object_ ([screenshot](https://i.imgur.com/m1EuG2P.jpg)). But it then prints an error that seems to indicate the type is _array_:  

`index.js:1 Warning: Failed prop type: Invalid prop `data` of type `array` supplied to `Results`, expected `object`.`

Why does this error seem to say that the prop's type is _array_?  Haven't I confirmed that the type of _object_?  What am I misunderstanding?

Thanks.

edit:  Sorry, right before I submitted this I saw a link on the page that read, "Similar to 4 other issues."  But before I could click it, the post submitted and the link disappeared. I'm happy to check those to see if they are related to my issue.

**Comment by ljharb** [comment]
The typeof an array is object.

PropTypes.object does not accept an array. You need PropTypes.array for that.

**Comment by cagross** [comment]
If I change my code from `PropTypes.object` to `PropTypes.array`, then the console error changes to:

`Warning: Failed prop type: Invalid prop `data` of type `object` supplied to `Results`, expected `array`.`

Thoughts on that?

**Comment by ljharb** [comment]
Yes, that seems to be the implication.

You can console.log `Array.isArray(data)` to verify that.

**Comment by cagross** [comment]
> You can console.log Array.isArray(data) to verify that.

This helped, thanks!  I was indeed passing an object once, and an array at another point.  We can consider this resolved.

**Comment by Raph2ll** [comment]
Hey @cagross, how did you solve this?

**Comment by cagross** [comment]
In this case, I'm pretty sure my issues stemmed from the fact that in my code, I've called my `Results` function twice.

* In the first call I passed to it a `data` parameter which was an object.  This did not throw any errors (as expected)
* In the second call I inadvertently passed to it a `data` parameter which was an array.  This threw an error.  

Once I tracked down both calls and actually inspected what was passed to each, I realized that indeed I was passing an array to the second call, instead of an object. I changed it to an object and the issue was fixed.
